extensions/cactus-plugin-htlc-coordinator/src/main/typescript/plugin-htlc-coordinator.ts

```diff
@@ -0,0 +1,353 @@
+}
+
+export class PluginHTLCCoordinator implements ICactusPlugin, IPluginWebService {
+  private readonly instanceId: string;
+  private readonly log: Logger;
+  private readonly pluginRegistry: PluginRegistry;
+  private endpoints: IWebServiceEndpoint[] | undefined;
+  private httpServer: Server | SecureServer | null = null;
+
+  public static readonly CLASS_NAME = "PluginHTLCCoordinator";
+
+  public get className(): string {
+    return PluginHTLCCoordinator.CLASS_NAME;
+  }
+
+  constructor(public readonly options: IPluginHTLCCoordinatorOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(options, `${fnTag} arg options`);
+    Checks.truthy(options.instanceId, `${fnTag} options.instanceId`);
+    Checks.truthy(options.pluginRegistry, `${fnTag} options.pluginRegistry`);
+
+    const level = this.options.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.instanceId = options.instanceId;
+    this.pluginRegistry = options.pluginRegistry;
+  }
+
+  public getInstanceId(): string {
+    return this.instanceId;
+  }
+
+  public getPackageName(): string {
+    return `@hyperledger/cactus-plugin-htlc-coordinator`;
+  }
+
+  public getHttpServer(): Optional<Server | SecureServer> {
+    return Optional.ofNullable(this.httpServer);
+  }
+
+  public async shutdown(): Promise<void> {
+    const serverMaybe = this.getHttpServer();
+    if (serverMaybe.isPresent()) {
+      const server = serverMaybe.get();
+      await promisify(server.close.bind(server))();
+    }
+  }
+
+  async registerWebServices(app: Express): Promise<IWebServiceEndpoint[]> {
+    const webServices = await this.getOrCreateWebServices();
+    await Promise.all(webServices.map((ws) => ws.registerExpress(app)));
+    return webServices;
+  }
+
+  public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+    if (Array.isArray(this.endpoints)) {
+      return this.endpoints;
+    }
+    const endpoints: IWebServiceEndpoint[] = [];
+    {
+      const endpoint = new OwnHTLCEndpoint({
+        coordinator: this,
+        logLevel: this.options.logLevel,
+      });
+      endpoints.push(endpoint);
+    }
+    {
+      const endpoint = new CounterpartyHTLCEndpoint({
+        coordinator: this,
+        logLevel: this.options.logLevel,
+      });
+      endpoints.push(endpoint);
+    }
+    {
+      const endpoint = new WithdrawCounterpartyEndpoint({
+        coordinator: this,
+        logLevel: this.options.logLevel,
+      });
+      endpoints.push(endpoint);
+    }
+    return endpoints;
+  }
+
+  public async ownHTLC(
+    ownHTLCRequest: OwnHTLCRequest,
+  ): Promise<InvokeContractV1Response> {
+    const fnTag = `${this.className}#ownHTLC()`;
+    const connector = this.pluginRegistry.plugins.find(
+      (plugin) => plugin.getInstanceId() == ownHTLCRequest.connectorId,
+    ) as PluginLedgerConnectorBesu;
+
+    const pluginRegistry = new PluginRegistry({});
```

Comment:
@jagpreetsinghsasan Please refactor so that the existing plugin registry is used to look up plugin instances instead of creating one from scratch (tests will have to be updated to pass in a plugin registry via the constructor as well - normally the API server takes care of this)